templates/install_puppet.bat.erb

```diff
@@ -23,7 +23,7 @@ REM This may fail on agents without pxp-agent, but since this is not
 REM run interactively and the next command sets ERRORLEVEL, it's OK.
 net stop pxp-agent
 
-start /wait msiexec.exe /qn /norestart /i "<%= @_msi_location %>" /l*vx "<%= @_logfile %>" PUPPET_MASTER_SERVER="<%= @_puppet_master %>"
+start /wait msiexec.exe /qn /norestart /i "<%= @_msi_location %>" /l*vx "<%= @_logfile %>" PUPPET_MASTER_SERVER="<%= @_puppet_master %>" PUPPET_AGENT_ENVIRONMENT="<%= @environment %>"
```

Comment:
`server_facts['environment']` would return the environment setting in the puppetserver(puppet-master)'s puppet.conf file and would be bad all around... 

For the purposes of discussion:
Assume the following, the agent configured environment is "testing", and the ENC override-ed environment is "lab",

The puppet catalog will compile under lab but there will be a warning about the requested environment being set to testing. This at least in my opinion is a bad behavior that should be corrected...

Interestingly, when the agent environment setting is `production` (default), and the enc override setting is `testing`, the agent has no issues with functioning (regardless of a local environment directory existing to match the ENC override environment). However if the agent environment setting is explicitly updated to match the enc enforced environment `testing` and the `{puppet_code_dir}/environments/testing` directory does not exist, then the agent will die and fail to run. 

So at a bare minimum this PR needs to add code to ensure that topscopped `$::environment` directory exists...

There has been an long lived bug (or in this case its kind of a feature) in 3.x which made it into 4.x which set local scoped `$environment` to Agent requested environment, and `$::environment` to the ENC override-ed environment setting. I have observed this with the hiera.yaml directory environments path directives. It _MAY_ be possible that this bug/feature also exists in the catalog compilation... However in 4.x we gained access to `$::agent_specified_environment` fact which would lets us effectively set the MSI Installer ARG to be that of the previous puppet.conf setting... Unfortunately that fact does not exist in 3.8... Unfortunately this is not something which we can easily test with rspec-puppet.

I will investigate this "feature" in 3.8 master agent config to see if it functions in a catalog the same way it does in the hiera.conf

Marking this as a WIP until I get those answers.